### PR TITLE
Never show expanded list for ID fields

### LINKED ISSUE #2308: [FR] Never show expanded list for ID fields
### Proposal Summary

Ids are listed when they are few. They do not convey much information while holding a large space. I propose "never show expanded list" feature. Ids can always be shown in the same manner (with a drop down menu, as in the case for large sample counts).

image (2)

### What areas of FiftyOne does this feature affect?

-   [x] App: FiftyOne application
-   [ ] Core: Core `fiftyone` Python library
-   [ ] Server: FiftyOne server

### PR DESCRIPTION
**Pull request by lanzhenw**
fixes #2308 

## What changes are proposed in this pull request?
ID list should always be displayed as a filter search, instead of expanded selection options.

## How is this patch tested? If it is not, please explain why.

Locally

### Is this a user-facing change that should be mentioned in the release notes?

-   [x] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

File: app/packages/core/src/components/Filters/CategoricalFilter.tsx [hunk 1]
```diff
@@ -109,7 +110,11 @@ const Wrapper = ({
   }));
   const skeleton = useRecoilValue(isKeypointLabel(path));
 
-  if (results.length <= CHECKBOX_LIMIT || skeleton) {
+  if (
+    (results.length <= CHECKBOX_LIMIT &&
+      !NEVEREXPAND_FIELDS.includes(name.toLowerCase())) ||
```
[the remark this answers is not in the document]
Comment (lanzhenw) [reply app/packages/core/src/components/Filters/CategoricalFilter.tsx]: Yes, I just refactored it.

**Review by ShawnOakley — APPROVED** [review]
LGTM!

File: app/packages/core/src/components/Filters/CategoricalFilter.tsx [hunk 2]
```diff
@@ -398,21 +403,24 @@ const CategoricalFilter = <T extends V = V>({
...
-              fontSize: "1rem",
-              width: "100%",
-            }}
-            containerStyle={{ borderBottomColor: color, zIndex: 1000 }}
-            toKey={({ value }) => String(value)}
-          />
-        )}
+        {results !== null &&
+          (results.length > CHECKBOX_LIMIT ||
+            NEVEREXPAND_FIELDS.includes(name.toLowerCase())) &&
```
[the remark this answers is not in the document]
Comment (brimoor) [reply app/packages/core/src/components/Filters/CategoricalFilter.tsx]: Agree that abstracting this check into a util function would be clean

**Review by brimoor — CHANGES_REQUESTED** [review]
@lanzhenw one point of clarification that I should have made more clear in the issue description: it would be nice to apply this logic to any field of type `ObjectIdField`. By convention, all `id` fields always have type `ObjectIdField`, but there could be other such fields as well.

For example:

```py
import fiftyone as fo
import fiftyone.zoo as foz

dataset = foz.load_zoo_dataset("quickstart")
view = dataset.to_patches("ground_truth")

print(view)
```

has two top-level `ObjectIDField`s:

```
Dataset:     quickstart
Media type:  image
Num patches: 1232
Patch fields:
    id:           fiftyone.core.fields.ObjectIdField
    sample_id:    fiftyone.core.fields.ObjectIdField
    filepath:     fiftyone.core.fields.StringField
    tags:         fiftyone.core.fields.ListField(fiftyone.core.fields.StringField)
    metadata:     fiftyone.core.fields.EmbeddedDocumentField(fiftyone.core.metadata.ImageMetadata)
    ground_truth: fiftyone.core.fields.EmbeddedDocumentField(fiftyone.core.labels.Detection)
View stages:
    1. ToPatches(field='ground_truth', config=None)
```

**Comment by brimoor** [comment]
Another point of discussion: one could argue that this "never expand" behavior is primarily important only for nested label attributes such as `ground_truth.detections.id` in the `quickstart` dataset (because multiple widgets appear altogether when the `ground_truth` field's widgets tray is expanded) and *not* the top-level `id` field, since each top-level field has its own expandable filter tray.

Curious what folks think about this?

**Comment by brimoor** [comment]
Just to clarify: we decided offline to force all `ObjectIdField` fields to use "non-expanded" mode, whether top-level or nested

File: app/packages/core/src/components/Filters/CategoricalFilter.tsx [hunk 3]
```diff
@@ -361,6 +362,7 @@ const CategoricalFilter = <T extends V = V>({
   const theme = useTheme();
   const field = useRecoilValue(fos.field(path));
   const countsLoadable = useRecoilValueLoadable(countsAtom);
+  const neverShowExpansion = NEVEREXPAND_FIELDS.includes(name.toLowerCase());
```
Comment (brimoor) [inline app/packages/core/src/components/Filters/CategoricalFilter.tsx:365]: `NEVEREXPAND_FIELDS` is not defined

**Review by brimoor — CHANGES_REQUESTED** [review]

**Review by brimoor — APPROVED** [review]
LGTM

**Pull request merged**